streampipes-client-python/streampipes_client/functions/function_handler.py

```diff
@@ -0,0 +1,192 @@
+
+logger = logging.getLogger(__name__)
+
+
+# TODO Exception should be removed once all brokers are implemented.
+class UnsupportedBroker(Exception):
+    """Exception if a broker isn't implemented yet."""
+
+    def __init__(self, message):
+        super().__init__(message)
+
+
+class FunctionHandler:
+    """The function handler manages the StreamPipesFunctions.
+    It controls the connection to the brokers, starts the functions, manages the broadcast of the live data
+    and is able to stop the connection to the brokers and functions.
+
+    Parameters
+    ----------
+    registration: Registration
+        The registration, that contains the StreamPipesFunctions.
+    client: StreamPipesClient
+        The client to interact with the API.
+    """
+
+    def __init__(self, registration: Registration, client: StreamPipesClient) -> None:
+        self.registration = registration
+        self.client = client
+        self.stream_contexts: Dict[str, DataStreamContext] = {}
+        self.brokers: List[Broker] = []
+
+    def initializeFunctions(self) -> None:
+        """Creates the context for every data stream and starts the event loop to manage the StreamPipesFunctions.
+
+        Returns
+        -------
+        None
+        """
+        # Choose the broker and collect the schema for every data stream
+        for streampipes_function in self.registration.getFunctions():
+            for stream_id in streampipes_function.requiredStreamIds():
+                # Get the data stream schema from the API
+                data_stream: DataStream = self.client.dataStreamApi.get(stream_id)  # type: ignore
+                # Get the broker
+                broker = self._get_broker(data_stream.event_grounding.transport_protocols[0].broker_hostname)
+                # Assign the functions, broker and schema to every stream
+                if stream_id in self.stream_contexts.keys():
+                    self.stream_contexts[stream_id].add_function(streampipes_function)
+                else:
+                    self.stream_contexts[stream_id] = DataStreamContext(
+                        functions=[streampipes_function], schema=data_stream, broker=broker
+                    )
+                logger.info(f"Using {broker.__class__.__name__} for {streampipes_function}")
+
+        # Start the function loop or add it as tasks if a loop is already running
+        try:
+            loop = asyncio.get_running_loop()
+        except RuntimeError:
+            asyncio.run(self._function_loop())
+        else:
+            loop.create_task(self._function_loop())
+
+    def _get_broker(self, broker_name: str) -> Broker:  # TODO implementation for more transport_protocols
+        """Get a broker by a name.
+
+        Parameters
+        ----------
+        broker_name: str
+            A string that represents a broker.
+
+        Returns
+        -------
+        The broker which belongs to the name.
+        """
+        if broker_name == "nats":
```

Comment:
Would you mind to create an enum for supported brokers?

Reply:
I can do this. That's a good idea